numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -20,12 +20,21 @@
 
 typedef struct NewNpyArrayIterObject_tag NewNpyArrayIterObject;
 
+char no_context_manager[] = "nditer has writebackifcopy semantics, "
+                "use as a context manager (see nditer help). "
+                "Will raise another deprecation warning in dealloc";
+
 struct NewNpyArrayIterObject_tag {
     PyObject_HEAD
     /* The iterator */
     NpyIter *iter;
     /* Flag indicating iteration started/stopped */
     char started, finished;
+    /* iter must used as a context manager if writebackifcopy semantics used */
+    #define ITR_INSIDE 1<<0
+    #define ITR_HAS_WRITEBACKIFCOPY 1<<1
+    #define ITR_EXITED 1<<2
```

Comment:
Why not an enum?

Reply:
Or even just a `char` for each flag - we don't care about efficiency here.

Reply:
fixed in 9a45b3d055